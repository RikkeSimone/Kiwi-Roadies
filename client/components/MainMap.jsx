import React from 'react'
import { Icon } from 'leaflet'
import { Map, TileLayer, Marker } from 'react-leaflet'
import { connect } from 'react-redux'

// coordinates to center map
const position = [-40.852931, 172.762057]

const carIcon = new Icon({
  iconUrl: './graphics/kiwiroadieslogo.png',
  iconSize: [50, 50]
})

const startPosition = [-36, 175]

// To figure out how to pass map props if using this one?
// https://github.com/arjunghimire/react-leaflet-routing-machine-example

const MainMap = ({ dataSet }) => (

  <div className='map-container'>
    <Map center={position} zoom={5} >
      <TileLayer
        url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
        attribution='&copy; <a href="http://osm.org/copyright">OpenStreetMap</a> contributors'
      />

      {/* <Routing
        // optional: map layer
        // mapLayer={
        //   name: 'OpenStreetMap',
        //   type: 'TileLayer',
        //   url: `https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png`,
        //   attribution: '&amp;copy <a href=&quot;http://osm.org/copyright&quot;>OpenStreetMap</a> contributors'
        // }

        // optional : initial region displayed
        initialRegion={startPosition}

        // optional: coordinates of the starting point
        from={startPosition}

        // optional: coordinates of the arriving point
        to={endPosition}
      /> */}

      <Marker
        position={startPosition}
        icon={carIcon}
      />

    </Map>
  </div>
)

const mapStateToProps = (state) => {
  console.log('state:', state.dataSet.trips[0])
  return {
    dataSet: state.dataSet
  }
}

export default connect(mapStateToProps)(MainMap)
