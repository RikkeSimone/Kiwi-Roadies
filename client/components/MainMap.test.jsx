import React from 'react'
import { renderWithRedux } from '../testing/utils'
import { screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import 'regenerator-runtime/runtime';

import MainMap from './MainMap'

test('MainMap contains a map', async () => {
  const { asFragment } = renderWithRedux(<MainMap />)
  expect(asFragment()).toMatchSnapshot()
})

