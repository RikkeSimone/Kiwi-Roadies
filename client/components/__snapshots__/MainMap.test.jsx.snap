// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainMap contains a map 1`] = `
<DocumentFragment>
  <div
    class="map-container"
  >
    <div
      class="leaflet-container leaflet-grab leaflet-touch-drag"
      style="position: relative;"
      tabindex="0"
    >
      <div
        class="leaflet-pane leaflet-map-pane"
        style="left: 0px; top: 0px;"
      >
        <div
          class="leaflet-pane leaflet-tile-pane"
        >
          <div
            class="leaflet-layer "
            style="z-index: 1;"
          >
            <div
              class="leaflet-tile-container leaflet-zoom-animated"
              style="z-index: 18; left: 0px; top: 0px;"
            >
              <img
                alt=""
                class="leaflet-tile"
                role="presentation"
                src="https://c.tile.openstreetmap.org/5/31/19.png"
                style="width: 256px; height: 256px; left: -91px; top: -252px;"
              />
            </div>
          </div>
        </div>
        <div
          class="leaflet-pane leaflet-shadow-pane"
        />
        <div
          class="leaflet-pane leaflet-overlay-pane"
        />
        <div
          class="leaflet-pane leaflet-marker-pane"
        >
          <img
            alt=""
            class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
            src="./graphics/kiwiroadieslogo.png"
            style="margin-left: -25px; margin-top: -25px; width: 50px; height: 50px; left: 51px; top: -141px; z-index: -141;"
            tabindex="0"
          />
        </div>
        <div
          class="leaflet-pane leaflet-tooltip-pane"
        />
        <div
          class="leaflet-pane leaflet-popup-pane"
        />
      </div>
      <div
        class="leaflet-control-container"
      >
        <div
          class="leaflet-top leaflet-left"
        >
          <div
            class="leaflet-control-zoom leaflet-bar leaflet-control"
          >
            <a
              aria-label="Zoom in"
              class="leaflet-control-zoom-in"
              href="#"
              role="button"
              title="Zoom in"
            >
              +
            </a>
            <a
              aria-label="Zoom out"
              class="leaflet-control-zoom-out"
              href="#"
              role="button"
              title="Zoom out"
            >
              −
            </a>
          </div>
        </div>
        <div
          class="leaflet-top leaflet-right"
        />
        <div
          class="leaflet-bottom leaflet-left"
        />
        <div
          class="leaflet-bottom leaflet-right"
        >
          <div
            class="leaflet-control-attribution leaflet-control"
          >
            <a
              href="https://leafletjs.com"
              title="A JS library for interactive maps"
            >
              Leaflet
            </a>
             | © 
            <a
              href="http://osm.org/copyright"
            >
              OpenStreetMap
            </a>
             contributors
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
